src/main/java/software/aws/neptune/gremlin/GremlinQueryExecutor.java

```diff
@@ -401,43 +402,73 @@ public ResultSet executeQuery(final String sql, final Statement statement) throw
+        Long unnamedColumnIndex = 0L;
         for (final Object result : results.stream().map(Result::getObject).collect(Collectors.toList())) {
-            if (!(result instanceof LinkedHashMap)) {
-                // Best way to handle it seems to be to issue a warning.
-                LOGGER.warn(String.format("Result of type '%s' is not convertible to a Map and will be skipped.",
-                        result.getClass().getCanonicalName()));
-                continue;
-            }
-
-            // We don't know key or value types, so pull it out raw.
-            final Map<?, ?> uncastedRow = (LinkedHashMap<?, ?>) result;
+            if (result instanceof LinkedHashMap) {
+                // We don't know key or value types, so pull it out raw.
+                final Map<?, ?> uncastedRow = (LinkedHashMap<?, ?>) result;
 
-            // Convert generic key types to string and insert in new map with corresponding value.
-            final Map<String, Object> row = new HashMap<>();
-            uncastedRow.forEach((key, value) -> row.put(key.toString(), value));
+                // Convert generic key types to string and insert in new map with corresponding value.
+                final Map<String, Object> row = new HashMap<>();
+                uncastedRow.forEach((key, value) -> row.put(key.toString(), value));
 
-            // Add row to List of rows.
-            rows.add(row);
+                // Add row to List of rows.
+                rows.add(row);
 
-            // Get columns from row and put in columns List if they aren't already in there.
-            for (final String key : row.keySet()) {
-                if (!columns.containsKey(key)) {
-                    final Object value = row.get(key);
-                    if (GremlinTypeMapping.checkContains(value.getClass())) {
-                        columns.put(key, value.getClass());
-                    } else {
+                // Get columns from row and put in columns List if they aren't already in there.
+                for (final String key : row.keySet()) {
+                    if (!columns.containsKey(key)) {
+                        final Object value = row.get(key);
+                        if (GremlinTypeMapping.checkContains(value.getClass())) {
+                            columns.put(key, value.getClass());
+                        } else {
+                            columns.put(key, String.class);
+                        }
+                    } else if (columns.get(key) != row.get(key)) {
                         columns.put(key, String.class);
                     }
-                } else if (columns.get(key) != row.get(key)) {
-                    columns.put(key, String.class);
                 }
+            } else if (GremlinTypeMapping.checkContains(result.getClass())) {
+                // Result is scalar - generate a new key for the column
+                String key = generateColumnKey(unnamedColumnIndex);
```

Comment:
I think we can simplify this block down a lot if the columns.containsKey(key) while loop is moved to a separate function.

We could do something this below:
`
long findNextValidColumnIndex(final long currentIndex) {
    long index = currentIndex;
    while (columns.containsKey(generateColumnKey(index++))) {
        if (unnamedColumnIndex == Long.MAX_VALUE) {
            LOGGER.warn(String.format("Reached the maximum number of column keys available for scalar columns: %d", currentIndex));
            throw SqlError.createSQLException(
                    LOGGER,
                    SqlState.NUMERIC_VALUE_OUT_OF_RANGE,
                    SqlError.INVALID_MAX_FIELD_SIZE);
            }
    return index;
}
`

Then we have the following in the for loop:
`
// Find next valid unnamed index and generate column key name for it.
unnamedColumnIndex = findNextValidColumnIndex(unnamedColumnIndex); 
final String key = generateColumnKey(unnamedColumnIndex);
columns.put(key, result.getClass());

// Create and add new row with generated key.
final Map<String, Object> row = new HashMap<>();
row.put(key, result);
rows.add(row);
`

Reply:
Let me know if you disagree - just a thought.l